### DISCUSSION TITLE
Download with TransferConfigs

### DISCUSSION THREAD

**Discussion by amircohere** [awslabs/aws-sdk-rust#964] [Q&A] [answered]
In python with Boto3, I can do the following:

```python
from boto3.s3.transfer import TransferConfig

config = TransferConfig(
    multipart_threshold=4 * 1024 * 1024 * 1024,  # 4GB
    max_concurrency=1,
    multipart_chunksize=32 * 1024 * 1024,  # 32MB
    num_download_attempts=299,
)

# some code here...

self.s3.download_file(
    Bucket="commoncrawl",
    Key="path_to_file.txt",
    Filename="local.txt",
    Config=config,
)
```
Is there any way to do this from aws-sdk-rust?

**Comment by jdisanti** [comment]
This functionality isn't implemented yet.

**Reply by brainstorm** [reply, marked as answer]
We @umccr need this for https://github.com/awslabs/aws-sdk-rust/issues/1136#issuecomment-2092006836, what would it take to up the prio from `p3` and have it in the [public project roadmap](https://github.com/orgs/awslabs/projects/50)?

EDIT: AWS team seems to be working on it internally with no ETA: https://github.com/smithy-lang/smithy-rs/pull/3622#issuecomment-2092957304
